File: benchmarks/python/array_iteration.py
#!/usr/bin/env python3
"""BENCH_ARRAY_SIZE=5000000"""


def sum_range(n: int) -> int:
    total = 0
    for i in range(n):
        total += i
    return total


def main() -> None:
    print(sum_range(5_000_000))


if __name__ == "__main__":
    main()

File: benchmarks/cpp/array_iteration.cpp
// BENCH_ARRAY_SIZE=5000000
#include <iostream>

static long long sum_range(int n) {
    long long total = 0;
    for (int i = 0; i < n; ++i) total += i;
    return total;
}

int main() {
    std::cout << sum_range(5000000) << '\n';
    return 0;
}
